core/mondoo-okta-security.mql.yaml

```diff
@@ -1,379 +1,1985 @@
+        ### User impact 
+
+        High
+
+        When signing in to their org, end users are prompted to enroll in required factors and may enroll in any factors set to optional. Factors that have been disabled aren't visible to end users.
+    refs:
+      - url: https://help.okta.com/en-us/Content/Topics/Security/healthinsight/strong-factors.htm
+        title: Disable weaker MFA factors in factor enrollment policies
+      - url: https://registry.terraform.io/providers/okta/okta/latest/docs/resources/policy_mfa_default
+        title: Terraform Registry - okta_policy_mfa_default 
+    variants:
+      - uid: mondoo-okta-security-disable-weaker-mfa-factors-in-factor-enrollment-policies-runtime
+      - uid: mondoo-okta-security-disable-weaker-mfa-factors-in-factor-enrollment-policies-terraform-hcl
+      - uid: mondoo-okta-security-disable-weaker-mfa-factors-in-factor-enrollment-policies-terraform-plan
+      - uid: mondoo-okta-security-disable-weaker-mfa-factors-in-factor-enrollment-policies-terraform-state
+  - uid: mondoo-okta-security-disable-weaker-mfa-factors-in-factor-enrollment-policies-runtime
+    title: Disable weaker MFA factors in factor enrollment policies
     impact: 80
+    filters: asset.platform == "okta-org"
     mql: |
-      okta.policies.signOn.where(_.rules.all(status == "ACTIVE")).all(_.rules.all(actions["signon"]["session"]["maxSessionIdleMinutes"] <= 120))
-      okta.policies.signOn.where(_.rules.all(status == "ACTIVE")).all(_.rules.all(actions["signon"]["session"]["maxSessionLifetimeMinutes"] <= 1440))
+      okta.policies.mfaEnroll.all( _.settings['factors']['okta_email']['enroll']['self'] != /(OPTIONAL|REQUIRED)/ )
+      okta.policies.mfaEnroll.all( _.settings['factors']['okta_sms']['enroll']['self'] != /(OPTIONAL|REQUIRED)/ )
+      okta.policies.mfaEnroll.all( _.settings['factors']['phone_number']['enroll']['self'] != /(OPTIONAL|REQUIRED)/ )
     docs:
-      desc: "The session lifetime determines the maximum idle time of a user's Okta session, and when the session expires. By enforcing a limited session lifetime for users, admins reduce the window of time wherein a malicious third party can try to access a user's applications in an active session.\n\nThis limitation is especially important considering the workforce is no longer strictly confined within the walls of an office. It minimizes the risk of session cookies misuse or hijacking.\n\nThe maximum session lifetime ensures that a session expires after the set amount of time, even if idle time never expires. The user session expires when the user has been inactive on Okta for the set time period, regardless of the maximum Okta session lifetime.\n"
-      audit: |
-        __Use `cnspec run`__
+      remediation: 
+        - id: console 
+          desc: | 
+            ### Enable strong factors for factor enrollment
+            1. In the Admin Console, go to SecurityMultifactor.
+            2. Click Factor Enrollment.
```

Comment:
```suggestion
            2. Select Factor Enrollment.
```

Reply:
@tas50 these are not our steps. they are from Okta's docs. They are copy/pasted.